File: 2.Warmup/2.9. Palindrome Number/palindrome-number.py
# Palindrome number - https://leetcode.com/problems/palindrome-number/description/
print("------ Palindrome number - https://leetcode.com/problems/palindrome-number/description/ ---------")

def palindrome_number(num):
    sign = -1 if num < 0 else 1
    num_copy = num
    num = sign * num
    rev = 0
    while num > 0:
        rem = num % 10
        rev = (rev * 10) + rem
        num = int(num/10)
    return rev == num_copy

print(palindrome_number(121))
print(palindrome_number(-121))
print(palindrome_number(123))

File: 2.Warmup/2.9. Palindrome Number/palindrome-number.js
//Palindrome number - https://leetcode.com/problems/palindrome-number/description/
console.log("------ Palindrome number - https://leetcode.com/problems/palindrome-number/description/ ---------");

function palindromeNumber(num){
    let sign = Math.sign(num)
    let initial_num = num
    num = sign * num
    let final = 0
    while(num > 0){
        let remainder = num % 10
        final = final * 10 + remainder
        num = Math.floor(num/10)
    }
    return (final) === initial_num
}

console.log(palindromeNumber(121))
console.log(palindromeNumber(-121))
console.log(palindromeNumber(123))